contrib/hashicorp/vault/vault.go

```diff
@@ -0,0 +1,98 @@
+// The vault package contains functions to construct an *http.Client that will
+// integrate with the github.com/hashicorp/vault/api and collect traces to send
+// to DataDog.
+//
+// The easiest way to use this package is to create an *http.Client with
+// NewHttpClient, and put it in the vault api config that's passed to the
+// NewClient function.
+//    import "github.com/hashicorp/vault/api"
+//    import trace "gopkg.in/DataDog/dd-trace-go.v1/contrib/hashicorp/vault"
+//
+//    func ... {
+//        api.NewClient(&api.Config{HttpClient: trace.NewHttpClient()})
+//    }
+//
+// If you are already using your own *http.Client with vault, you can use the
+// WrapHttpTransport function to wrap the client with the tracer code. Your
+// *http.Client will continue to work as before, but will also capture traces.
+//
+//    func ... {
+//        var myHttpClient *http.Client
+//        myHttpClient = ...
+//        api.NewClient(&api.Config{HttpClient: trace.WrapHttpTransport(myHttpClient)})
+//    }
+//
+package vault
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	httptrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+
+	vault "github.com/hashicorp/vault/api"
```

Comment:
Why are we renaming this to `vault`? I don't think there is a point unless it collides with another `api` package.